### PR TITLE
Fix file handle leak in Helpers.cpp

### PR DESCRIPTION
**Pull request by xiaoyinl**
We need to make sure `fclose(file)` is always called after the file has been successfully opened, so move it after `Error` label.

Another change is to move error message printing outside `#ifdef _WIN32`, so that it is printed on Linux.

### CODE REVIEW

File: bin/ch/Helpers.cpp [hunk 1]
```diff
@@ -370,7 +370,7 @@ HRESULT Helpers::LoadBinaryFile(LPCSTR filename, LPCSTR& contents, UINT& lengthB
             }
 #endif
             fprintf(stderr, "\n");
-            IfFailGo(E_FAIL);
+            return E_FAIL;
```
Comment (kfarnung) [inline bin/ch/Helpers.cpp]: Looks like the `return E_FAIL` can be moved out a level and eliminate the `else` condition?

Comment (xiaoyinl) [inline bin/ch/Helpers.cpp]: Yes, you're right. I didn't notice this. I will fix it soon.

Comment (xiaoyinl) [inline bin/ch/Helpers.cpp]: Done. Thanks!

**Comment by kfarnung** [comment]
Once all the checks pass I'll merge this PR.

**Pull request merged**